lib/capability/capability.go

```diff
@@ -25,12 +25,17 @@ func (caps capabilitiesSort) Len() int           { return len(caps) }
 func (caps capabilitiesSort) Swap(i, j int)      { caps[i], caps[j] = caps[j], caps[i] }
 func (caps capabilitiesSort) Less(i, j int) bool { return string(caps[i]) < string(caps[j]) }
 
-// SetCapabilities populates and returns cluster capabilities from ClusterVersion capabilities spec.
-func SetCapabilities(config *configv1.ClusterVersion) ClusterCapabilities {
+// SetCapabilities populates and returns cluster capabilities from ClusterVersion capabilities spec. This method also
+// ensures that no previousily enabled capability is now disabled and returns any such (tainted-in) capabilities.
+func SetCapabilities(config *configv1.ClusterVersion,
+	existingEnabled map[configv1.ClusterVersionCapability]struct{}) (ClusterCapabilities, []configv1.ClusterVersionCapability) {
```

Comment:
Why the separate return field, instead of adding this as a new property in `ClusterCapabilities`, or converting `ClusterCapabilities.EnabledCapabilities` to a `map[configv1.ClusterVersionCapability]bool` for "tainted vs. requested"?

Reply:
The reason I changed this and removed the `bool` was so I could easily compare for update. I can then use `equalSyncWork` and fold capabilities in just like overrides. The other option of course is to have a capabilities specific equals method which gets called from `equalSyncWork`, which ignores the `bool`.